src/main/java/io/stargate/sgv2/jsonapi/service/shredding/model/JsonExtensionType.java

```diff
@@ -0,0 +1,59 @@
+package io.stargate.sgv2.jsonapi.service.shredding.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Types that "extend" JSON scalar type: by wrapping scalar value (usually String) in a single-entry
+ * JSON Object value where key indicates the type. These are generally used to retain client-side
+ * type information in cases where processing on server-side is still based on enclosed String
+ * value; one exception being "EJSON Date" which is used to specify Date values for indexing and
+ * sort purposes.
+ */
+public enum JsonExtensionType {
+  EJSON_DATE("$date"),
+
+  OBJECT_ID("$objectId"),
+
+  UUID("$uuid"),
+
+  UUID_V6("$uuid6"),
```

Comment:
These are covered in the spec (not sure if you have access, can share on slack), but basically we support different generation methods. I'll check with Aaron whether we need separation for wrappers tho...

As to JSON Object notation: it is important to keep all extension types expressed as 1-element JSON Object. So I think separating subtype into different field would complicate design a lot without bringing many benefits.

Reply:
FWTW, I think I'll bring up suggestion that in Document only `$uuid` should be used, and allow any valid UUID value to be inserted/modified.

But `createCollection()` would accept wider range to specify which version is auto-generated: in particular, versions 6 and 7 are quite nice as they have time(stamp)-based lexicographic sorting (v1 alas does not despite containing all bits that would allow it). So it's possibly to have sort of time-ordered keys for a Collection.